extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/media/GoogleMediaImporter.java

```diff
@@ -0,0 +1,504 @@
+import org.datatransferproject.spi.transfer.types.PermissionDeniedException;
+import org.datatransferproject.transfer.ImageStreamProvider;
+import org.datatransferproject.types.common.models.media.MediaAlbum;
+import org.datatransferproject.types.common.models.media.MediaContainerResource;
+import org.datatransferproject.types.common.models.photos.PhotoModel;
+import org.datatransferproject.types.common.models.videos.VideoModel;
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+
+public class GoogleMediaImporter implements Importer<TokensAndUrlAuthData, MediaContainerResource> {
+
+  private final GoogleCredentialFactory credentialFactory;
+  private final JobStore jobStore;
+  private final JsonFactory jsonFactory;
+  private final ImageStreamProvider imageStreamProvider;
+  private final Monitor monitor;
+  private final double writesPerSecond;
+  private final Map<UUID, GooglePhotosInterface> photosInterfacesMap;
+  private final GooglePhotosInterface photosInterface;
+
+  public GoogleMediaImporter(
+      GoogleCredentialFactory credentialFactory,
+      JobStore jobStore,
+      JsonFactory jsonFactory,
+      Monitor monitor,
+      double writesPerSecond) {
+    this(
+        credentialFactory,
+        jobStore,
+        jsonFactory,
+        new HashMap<>(),
+        null,
+        new ImageStreamProvider(),
+        monitor,
+        writesPerSecond);
+  }
+
+  @VisibleForTesting
+  GoogleMediaImporter(
+      GoogleCredentialFactory credentialFactory,
+      JobStore jobStore,
+      JsonFactory jsonFactory,
+      Map<UUID, GooglePhotosInterface> photosInterfacesMap,
+      GooglePhotosInterface photosInterface,
+      ImageStreamProvider imageStreamProvider,
+      Monitor monitor,
+      double writesPerSecond) {
+    this.credentialFactory = credentialFactory;
+    this.jobStore = jobStore;
+    this.jsonFactory = jsonFactory;
+    this.photosInterfacesMap = photosInterfacesMap;
+    this.photosInterface = photosInterface;
+    this.imageStreamProvider = imageStreamProvider;
+    this.monitor = monitor;
+    this.writesPerSecond = writesPerSecond;
+  }
+
+  @Override
+  public ImportResult importItem(
+      UUID jobId,
+      IdempotentImportExecutor idempotentImportExecutor,
+      TokensAndUrlAuthData authData,
+      MediaContainerResource data)
+      throws Exception {
+    if (data == null) {
+      // Nothing to do
+      return ImportResult.OK;
+    }
+
+    // Uploads album metadata
+    if (data.getAlbums() != null && data.getAlbums().size() > 0) {
+      for (MediaAlbum album : data.getAlbums()) {
+        idempotentImportExecutor.executeAndSwallowIOExceptions(
+            album.getId(), album.getName(), () -> importSingleAlbum(jobId, authData, album));
+      }
+    }
+
+    long bytes = importPhotos(data.getPhotos(), idempotentImportExecutor, jobId, authData);
+    bytes += importVideos(data.getVideos(), idempotentImportExecutor, jobId, authData);
```

Comment:
We assume that `MediaContainerResource` has either photos or videos and process them at the same time. 
Is it possible that a single `MediaContainerResource` has both?